Reduce CI vs local test environment inconsistencies

kylelaker-ccpo opened the pull request:
Addresses common causes of tests failing in the CI pipeline that do not fail locally, including out-of-sync coverage exclusions and static asset generation. I've attempted to add comments to prevent similar issues from occurring in the future.

Ticket: [AT-5996](https://ccpo.atlassian.net/browse/AT-5996)

kylelaker-ccpo commented:
Alright barring any other merges into `develop`, this `git pull --rebase origin develop && git push --force` should be the last one; should be ready to review.

jasonburkert-ccpo commented:
> git pull --rebase origin develop && git push --force

Thanks for that.  Will shepherd.

kylelaker-ccpo commented:
Looks like maybe during your push, for some reason, one of my later changes got lost (the one that added `pkg-config` to the list of `apt-get install`s). Went ahead and squashed your comment fixup into my first commit, rebased the series on top of `develop` and force-pushed again. Hopefully we're back to having the tests pass now :) -- Should find out in ~15min
